src/parser/warrior/arms/modules/core/Slam.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+import { formatPercentage } from 'common/format';
+import SpellLink from 'common/SpellLink';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import SpellUsable from 'parser/warrior/arms/modules/features/SpellUsable';
+
+class Slam extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+
+  badCast = 0;
+  totalCast = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SLAM), this._badCast);
```

Comment:
This has been a hot topic recently on Discord (naming the event listeners) and there's not a definitive answer, but I would suggest a more descriptive name, perhaps `onBadSlamCast` or `onSlamCast`, `handleSlam` etc., `_badCast` doesn't really tell much at first glance.

Reply:
Yeah i've seen the discussion, i have to admit, didn't knew how i should name this to be consistent :p